Wiki restructuring - The context (#1223)
Azure/ResourceModules: docs/wiki/The context.md

> This section aims to set the stage for the Common Azure Resource Modules Library (CARML) context and the motivations behind this initiative.

Reviewer: ```suggestion
This section explains the context of the Common Azure Resource Modules Library (CARML) and the motivation for its creation.
```
